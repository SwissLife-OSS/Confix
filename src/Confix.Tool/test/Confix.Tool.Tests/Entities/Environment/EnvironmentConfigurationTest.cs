using System.Text.Json.Nodes;
using Confix.Tool.Abstractions;

namespace ConfiX.Entities.Component.Configuration;

public class EnvironmentConfigurationTest : ParserTestBase
{
    [Fact]
    public void Parse_Should_BeValid()
    {
        ExpectValid(
            """
            {
                "name": "dev",
                "excludeFiles": [
                    "file1",
                    "file2"
                ]
            }
        """);
    }

    [Fact]
    public void Parse_Should_BeValid_When_OnlyName()
    {
        ExpectValid("\"dev\"");
    }

    [Fact]
    public void Parse_Should_BeValid_When_ExcludeFilesIsMissing()
    {
        ExpectValid(
            """
            {
                "name": "dev"
            }
        """);
    }

    [Fact]
    public void Parse_Should_BeValid_WhenEmpty()
    {
        ExpectValid(""" { } """);
    }

    public override object Parse(JsonNode json)
        => EnvironmentConfiguration.Parse(json);
}
