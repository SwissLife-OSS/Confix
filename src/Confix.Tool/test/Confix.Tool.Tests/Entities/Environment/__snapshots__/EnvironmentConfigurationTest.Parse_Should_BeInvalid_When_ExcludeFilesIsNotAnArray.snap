{
  "Message": "Expected array but got value",
  "Path": "$"
}
